apis/core/v1alpha2/readinessprovider_webhook.go

```diff
@@ -4,21 +4,29 @@
 package v1alpha2
 
 import (
+	"context"
 	"fmt"
 
+	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/util/validation/field"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
-// log is for logging in this package.
-var readinessproviderlog = logf.Log.WithName("readinessprovider-resource").WithValues("apigroup", "core")
+var (
+	// log is for logging in this package.
+	readinessproviderlog = logf.Log.WithName("readinessprovider-resource").WithValues("apigroup", "core")
+	// cli is the client used for making calls to the k8s API server.
+	cli client.Client
+)
 
 // SetupWebhookWithManager adds the webhook to the manager.
 func (r *ReadinessProvider) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	cli = mgr.GetClient()
```

Comment:
This can have a race condition if the manager isn't started before this is invoked. Is that possible? In other web hooks we have created a new client here and added APIs to the scheme.

Reply:
Refactored to creating a new client.